Follow contact permissions when importing contacts

fedys opened the pull request:
<!-- ## Which branch should I use for my PR?

Assuming that:

a = current major release
b = current minor release
c = future major release

* a.x for any features and enhancements (e.g. 5.x)
* a.b for any bug fixes (e.g. 4.4, 5.1)
* c.x for any features, enhancements or bug fixes with backward compatibility breaking changes (e.g. 5.x) -->

| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | 🟢 <!-- Use emojis to indicate positive (green) or negative (red) for each item in the table. -->
| New feature/enhancement? (use the a.x branch)      | 🔴
| Deprecations?                          | 🔴
| BC breaks? (use the c.x branch)        | 🔴
| Automated tests included?              | 🟢 <!-- All PRs must maintain or improve code coverage -->
| Related user documentation PR URL      |  <!-- required for new features -->
| Related developer documentation PR URL |  <!-- required for developer-facing changes -->
| Issue(s) addressed                     | <!-- prefix each issue number with "Fixes #", no need to create an issue if none exists, explain below instead -->

<!--
Additionally (see https://contribute.mautic.org/contributing-to-mautic/developer/code/pull-requests#work-on-your-pull-request):
 - Always add tests and ensure they pass.
 - Bug fixes must be submitted against the lowest maintained branch where they apply
   (lowest branches are regularly merged to upper ones so they get the fixes too.)
 - Features and deprecations must be submitted against the "4.x" branch.
-->

## Description

This PR adds permission checks during contact and company imports.

<!--
Please write a short README for your feature/bugfix. This will help people understand your PR and what it aims to do. If you are fixing a bug and if there is no linked issue already, please provide steps to reproduce the issue here.
-->
<!-- Remove HTML comment markup below to use the table for screenshots when relevant. -->
<!--
| Before                                 | After
| -------------------------------------- | ---
|                                        | 
-->


---
### 📋 Steps to test this PR:

1. Create a role that does not have permissions for editing contacts but has permissions to create contacts.
2. Login as a user that has the above role assigned to them.
3. Create a new contact import that contains rows for both inserts and updates.
4. Wait till the import is `imported`.
5. Any CSV rows containing updates should be ignored. 
6. There should be error messages like `User '<username>' has insufficient permissions` for all the rows containing updates.
7. Validate that permissions are checked against a user who **last updated a given import**.
8. Validate the above steps for company imports at `/s/companies/import/new`. The company import follows the same `Contact Permissions`. There are no separate `Company Permissions`.

<!--
If you have any deprecations and backwards compatibility breaks, list them here along with the new alternative.
-->

RCheesley committed:
Update to use default release notes, remove file (#10934)

RCheesley committed:
Fix issue with changelog (#11012)

volha-pivavarchyk committed:
Fix issue with access to contact information

volha-pivavarchyk committed:
Fix issue with access to contact information

volha-pivavarchyk committed:
Fix phpstan issues

volha-pivavarchyk committed:
Remove small changes

fedys committed:
Check permissions during contact import

fedys committed:
Username mentioned in the unauthorized message

fedys committed:
Use Company::getPermissionUser() for checking edit permissions in CompanyController::editAction()

fedys committed:
Check permissions during company import

fedys committed:
Use UserModel to set user's active permissions

fedys committed:
Test fixed

fedys committed:
A functional test

fedys committed:
Do not update company if there are no fields for update

shinde-rahul committed:
Added missing changes

shinde-rahul committed:
Added Campaign Controller test

shinde-rahul committed:
Added more coverage

shinde-rahul committed:
Update app/bundles/EmailBundle/Tests/Functional/EmailContactGridTest.php

JonasLudwig1998 committed:
Fixing bug in recent activity widget

Hugo-Prossaird committed:
style: update decision/action panel colors

Hugo-Prossaird committed:
feature: use Helper Button in segment's filters

driskell committed:
chore: Resolve paste event showing in campaign builder when nothing to paste

JonasLudwig1998 committed:
Fixing marketplace table so that select all works

Hugo-Prossaird committed:
fix: add ghost class on inactive button, primary class on active button

kuzmany committed:
fix: Correct campaign unpublished translation key in notification subject

JonasLudwig1998 committed:
Fixing problem with single search operators in search fields

JonasLudwig1998 committed:
Making sure operators are only treated as operators when written in uppercase

fedys committed:
CS fixes

fedys committed:
PHPStan fixes

fedys commented:
@escopecz The failing tests stop failing with `RuntimeException: Import does not have "modifiedBy" property set.` once https://github.com/acquia/mc-cs/pull/1534 is pushed.

Hugo-Prossaird committed:
refactoring: Use utility class

Co-authored-by: Anderson J. Eccel <116097999+andersonjeccel@users.noreply.github.com>

Hugo-Prossaird committed:
refactoring: Remove duplicate css (set by utility class)

Co-authored-by: Anderson J. Eccel <116097999+andersonjeccel@users.noreply.github.com>

Hugo-Prossaird committed:
fix: remove unexpected margin

escopecz commented:
This will be ready for testing once https://github.com/mautic/mautic/pull/14433 is merged and this PR rebased.

mzagmajster committed:
Wrap renderView method to handle tpl overwrites

eventAwareRenderView: We added this method, because ajax call needs to dispatch event to give plugins chance to replace templates.

driskell committed:
fix: The VIEW_INJECT_CUSTOM_CONTENT event was not getting dispatched with a valid viewName during AJAX content updates

driskell committed:
fix: Fix broken layout of My Categories on preference page (make them match public segments list)

driskell committed:
fix: Fix category selection on admin preference panel and segment filter only showing first 10

driskell committed:
fix: Doctrine migrations no longer result in No Active Transaction errors after DDL statements when using PHP 8

biozshock committed:
Add missing "isIndexed" and "charLegthLimit" fields to the API response of Contact Fields.

JonasLudwig1998 committed:
Getting rid of issues with defined but empty log.route

JonasLudwig1998 committed:
Fix SearchStringHelper.php so that tests in CI workflow won't fail

levente999 committed:
Point Triggers broken: Triggers vanish after first invocation #13817

JonasLudwig1998 committed:
Implementing change request in recentactivity.html.twig

JonasLudwig1998 committed:
Implementing change request in recentactivity.html.twig

JonasLudwig1998 committed:
Implementing change request in recentactivity.html.twig

LordRembo committed:
add allowed protocols for links in CK5, so people can add phone links (and some others)

LordRembo committed:
Update grapesjs (minor version) + grapesjs-mjml (bugfixes)

Hugo-Prossaird committed:
fix: allow preview email by contact for non-admin user

driskell committed:
fix: Move @defaults theme contents to assets and CoreBundle to ensure accessible in composer installation

driskell committed:
fix: Fix upsert sometimes storing incorrect ID in entity from a previous upsert operation

driskell committed:
fix: Creating or updating a contact via the Rest API discards seconds for date time fields

bastolen committed:
Fix template block so that the campaign editor has the correct format again

patrykgruszka committed:
fix: [DPMMA-3048] campaign change membership lead detach

patrykgruszka committed:
fix: [DPMMA-3048] revert changes in AbstractCampaignCommand

patrykgruszka committed:
fix: [DPMMA-3048] phpstan

andersonjeccel committed:
fix: Confirmation text being escaped incorrectly

escopecz committed:
Fixing the right menu height on smaller screens

fedys commented:
I closed this PR in favor of https://github.com/mautic/mautic/pull/14554. There were Git issues after merging `6.x`.
